src/server/connection.cpp

```diff
@@ -17,8 +17,9 @@ namespace osrm
 namespace server
 {
 
-Connection::Connection(boost::asio::io_service &io_service, RequestHandler &handler)
-    : strand(io_service), TCP_socket(io_service), timer(io_service), request_handler(handler)
+Connection::Connection(boost::asio::io_context &io_context, RequestHandler &handler)
+    : strand(boost::asio::make_strand(io_context)), TCP_socket(strand), timer(strand),
```

Comment:
The important change is here. Instantiating the socket and timer with the strand.